.github/workflows/build-android.yml

```diff
@@ -0,0 +1,55 @@
+name: Test Android Pull Requests
+
+on:
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'android/**'
+
+  push:
+    branches: [ main ]
+    paths:
+      - 'android/**'
+
+  # This allows us to manually run this job
+  workflow_dispatch:
+
+jobs:
+
+  kotlin-code-checks:
+    name: Code Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: gradle
+
+      - name: Grant execute permission for gradlew (Mobile)
+        working-directory: android/mobile
+        run: chmod +x gradlew
+
+      - name: Grant execute permission for gradlew (TV Custom)
+        working-directory: android/tv-custom
+        run: chmod +x gradlew
+
+      - name: Grant execute permission for gradlew (TV Device Auth Grant)
+        working-directory: android/tv-device-auth-grant
+        run: chmod +x gradlew
+
+      - name: Build with Gradle (Mobile)
```

Comment:
Use the `setup-gradle` action. It adds some nice caching.

https://github.com/gradle/actions?tab=readme-ov-file#the-setup-gradle-action